crates/ln-dlc-node/src/ln/app_event_handler.rs

```diff
@@ -99,12 +102,37 @@ where
                 temporary_channel_id,
                 ..
             } => {
+                // Fetching the originally announced channel, so we can reuse the `user_channel_id`.
+                // Note this should always be set, as the user must prepare the payment before
+                // funding the wallet.
+                // This will allow us to use the same `user_channel_id` on the app side as on the
+                // coordinator side. Unfortunately we have to fetch the `user_channel_id` as it is
+                // not provided in the `Event::OpenChannelRequest`.
+                let channel = self
+                    .node
+                    .storage
+                    .get_announced_channel(counterparty_node_id)?;
+
+                let user_channel_id = match channel {
+                    Some(mut channel) => {
+                        channel.channel_state = ChannelState::Pending;
+
+                        if let Err(e) = self.node.storage.upsert_channel(channel.clone()) {
+                            tracing::error!("Failed to update channel. Error: {e:#}");
+                        }
+
+                        channel.user_channel_id
```

Comment:
Nit: you could `clone` the `channel.user_channel_id` before and then _move_ the `channel` into `upsert_channel` function. This might be more efficient resource wise.